pytorch_lightning/callbacks/base.py

```diff
@@ -273,10 +273,19 @@ def on_save_checkpoint(
         """
         pass
 
-    def on_load_checkpoint(self, callback_state: Dict[str, Any]) -> None:
+    def on_load_checkpoint(
+        self, trainer: 'pl.Trainer', pl_module: 'pl.LightningModule', callback_state: Dict[str, Any]
+    ) -> None:
         """Called when loading a model checkpoint, use to reload state.
 
+        .. note::
+
+            The ``on_load_checkpoint`` will be called only if a state is provided.
+            Therefore, ``on_save_checkpoint`` hook need to be overridden to return a state.
```

Comment:
```suggestion
        Note:
            The ``on_load_checkpoint`` will be called only if a state is provided.
            Therefore, ``on_save_checkpoint`` hook need to be overridden to return a state.
```
can the note go at the bottom? I think it's best if args docs always come first

Reply:
Also I don't understand the second sentence:

> Therefore, ``on_save_checkpoint`` hook need to be overridden to return a state.

I think what you are trying to say relates more to special cases when the hook is used for more than just reloading state.

Reply:
Yes. Exactly.

Reply:
Reformulated :)